### PR TITLE
Fix invalid session ID in post roll ad sessions

### PR DESCRIPTION
**Pull request by rolandkakonyi**
### Problem

When playing a post-roll ad, "invalid session ID" error shows up on Conviva Touchstone.

### Solution

Postponing ending the session from `onPlaybackFinished` event with a delay of `0.1s` to check if there is an `onAdBreakStarted` event is received and then only end the session once `onAdBreakFinished` event is seen fixes the problem.

### Notes

Both CSAI and SSAI ad integration were adjusted to address the problem overall.

### Checklist
- [x] I added an update to `CHANGELOG.md` file - already added in #81
- [x] I ran all the tests in the project and they succeed

### CODE REVIEW

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 1]
```diff
@@ -592,12 +606,20 @@ extension ConvivaAnalytics: BitmovinPlayerListenerDelegate {
...
+        }
+        self.playbackFinishedDispatchWorkItem = playbackFinishedDispatchWorkItem
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.10, execute: playbackFinishedDispatchWorkItem)
     }
 
     func onStallStarted() {
         isStalled = true
-        DispatchQueue.main.asyncAfter(deadline: .now() + 0.10) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.10) { [weak self] in
+            guard let self else { return }
```
Comment (rolandkakonyi) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:622]: sneak: fixed missing weak reference

File: BitmovinConvivaAnalytics/Classes/Helper/BitmovinPlayerListener.swift [hunk 2]
```diff
@@ -33,8 +33,8 @@ extension BitmovinPlayerListener: PlayerListener {
         // The default SDK error handling is that it triggers the onSourceUnloaded before the onError event.
         // To track errors to conviva we need to delay the onSourceUnloaded to ensure the onError event is
         // called first.
-        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-            self.delegate?.onSourceUnloaded()
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) { [weak self] in
```
Comment (rolandkakonyi) [inline BitmovinConvivaAnalytics/Classes/Helper/BitmovinPlayerListener.swift:36]: sneak: fixed missing weak reference

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 3]
```diff
@@ -592,12 +606,20 @@ extension ConvivaAnalytics: BitmovinPlayerListenerDelegate {
...
         onPlaybackStateChanged(playerState: .CONVIVA_STOPPED)
-        internalEndSession()
+        let playbackFinishedDispatchWorkItem = DispatchWorkItem { [weak self] in
+            guard let self,
+                  let currentWorkItem = self.playbackFinishedDispatchWorkItem,
+                  !currentWorkItem.isCancelled else { return }
+            self.internalEndSession()
+        }
+        self.playbackFinishedDispatchWorkItem = playbackFinishedDispatchWorkItem
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.10, execute: playbackFinishedDispatchWorkItem)
```
Comment (stonko1994) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:616]: Nit:

```suggestion
        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(100), execute: playbackFinishedDispatchWorkItem)
```

or

```suggestion
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1, execute: playbackFinishedDispatchWorkItem)
```

**Review by stonko1994 — APPROVED** [review]

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 3 above)
Comment (rolandkakonyi) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: fixed in 865222d

**Pull request merged**